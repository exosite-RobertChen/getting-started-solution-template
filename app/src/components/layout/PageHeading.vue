<template>
  <div
    class="mt-4 pr-2 pl-6 page-heading"
    :class="{ 'mb-2': $vuetify.breakpoint.xs, 'mb-4': !$vuetify.breakpoint.xs }"
  >
    <h1 class="hidden-sm-and-up headline mt-4">{{ name }}</h1>
    <v-row align="center">
      <h1 class="hidden-xs-only mr-5">{{ name }}</h1>
      <slot />
    </v-row>
  </div>
</template>

<script>
export default {
  props: {
    name: {
      required: true,
      type: String
    }
  }
}
</script>

<style lang="scss">
.page-heading {
  h1 {
    font-weight: 300;
  }

  @media only screen and (min-width: 601px) {
    min-height: 60px;
    display: flex;
    align-items: flex-end;
  }
}
</style>
