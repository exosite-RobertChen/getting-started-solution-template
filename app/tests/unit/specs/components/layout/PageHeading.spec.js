import merge from 'lodash.merge'
import { shallowComponent } from '@/../tests/unit/common'
import PageHeading from '@/components/layout/PageHeading.vue'

const defaultProps = {
  propsData: {
    name: 'Devices'
  }
}

const factory = overrides =>
  shallowComponent(PageHeading, merge(defaultProps, overrides))

describe('PageHeading.vue', () => {
  const wrapper = factory()

  test('can be mounted', () => {
    expect(wrapper.contains(PageHeading)).toBe(true)
  })

  test('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
