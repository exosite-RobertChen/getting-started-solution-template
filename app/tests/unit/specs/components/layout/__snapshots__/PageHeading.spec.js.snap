// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeading.vue matches snapshot 1`] = `
<div
  class="mt-4 pr-2 pl-6 page-heading mb-4"
>
  <h1
    class="hidden-sm-and-up headline mt-4"
  >
    Devices
  </h1>
   
  <v-row-stub
    align="center"
    tag="div"
  >
    <h1
      class="hidden-xs-only mr-5"
    >
      Devices
    </h1>
     
  </v-row-stub>
</div>
`;
